cudamapper/tests/Test_CudamapperIndexBatcher.cu

```diff
@@ -0,0 +1,916 @@
+
+    // target FASTA: 20_reads.fasta
+    // read_0:  4 basepairs
+    // read_1:  6 basepairs
+    // read_2:  7 basepairs
+    // read_3:  4 basepairs
+    // read_4:  3 basepairs
+    // read_5:  8 basepairs
+    // read_6:  6 basepairs
+    // read_7:  3 basepairs
+    // read_8:  3 basepairs
+    // read_9:  5 basepairs
+    // read_10: 7 basepairs
+    // read_11: 3 basepairs
+    // read_12: 2 basepairs
+    // read_13: 4 basepairs
+    // read_14: 4 basepairs
+    // read_15: 2 basepairs
+    // read_16: 5 basepairs
+    // read_17: 6 basepairs
+    // read_18: 2 basepairs
+    // read_19: 4 basepairs
+    // max basepairs in index: 10
+    // indices: {0, 2}, {2, 1}, {3, 2}, {5, 1}, {6, 2}, {8, 2}, {10, 2}, {12, 3}, {15, 2}, {17, 2}, {19, 1}
+    // query_indices_per_host_batch: 5
+    // query_indices_per_device_batch: 2
+    // host batches: {{0, 2}, {2, 1}, {3, 2}, {5, 1}, {6, 2}}, {{8, 2}, {10, 2}, {12, 3}, {15, 2}, {17, 2}}, {{19, 1}}
+    // device bathes: {{0, 2}, {2, 1}}, {{3, 2}, {5, 1}}, {{6, 2}}
+    //                {{8, 2}, {10, 2}}, {{12, 3}, {15, 2}}, {{17, 2}}
+    //                {{19, 1}}
+
+    const number_of_indices_t query_indices_per_host_batch                    = 2;
+    const number_of_indices_t query_indices_per_device_batch                  = 1;
+    const number_of_indices_t target_indices_per_host_batch                   = 5;
+    const number_of_indices_t target_indices_per_device_batch                 = 2;
+    const std::shared_ptr<const claragenomics::io::FastaParser> query_parser  = claragenomics::io::create_kseq_fasta_parser(std::string(CUDAMAPPER_BENCHMARK_DATA_DIR) + "/10_reads.fasta", 1, false);
+    const std::shared_ptr<const claragenomics::io::FastaParser> target_parser = claragenomics::io::create_kseq_fasta_parser(std::string(CUDAMAPPER_BENCHMARK_DATA_DIR) + "/20_reads.fasta", 1, false);
+    const number_of_basepairs_t query_basepairs_per_index                     = 10;
+    const number_of_basepairs_t target_basepairs_per_index                    = 10;
+    const bool same_query_and_target                                          = false;
+
+    std::vector<BatchOfIndices> expected_batches;
+    {
+        // query 0, target 0
+        std::vector<IndexDescriptor> host_batch_query_indices{{0, 3}, {3, 3}};
+        std::vector<IndexDescriptor> host_batch_target_indices{{0, 2}, {2, 1}, {3, 2}, {5, 1}, {6, 2}};
+        IndexBatch host_batch{std::move(host_batch_query_indices),
+                              std::move(host_batch_target_indices)};
+        std::vector<IndexBatch> device_batches;
+        {
+            // device query 0, device target 0
+            std::vector<IndexDescriptor> device_batch_query_indices{{0, 3}};
+            std::vector<IndexDescriptor> device_batch_target_indices{{0, 2}, {2, 1}};
+            device_batches.push_back({std::move(device_batch_query_indices),
+                                      std::move(device_batch_target_indices)});
+        }
```

Comment:
Optional: This file is rather long, so I think it would increase readability if you just wrote
```
device_batches.push_back(
    {{0, 3}}, // query indicies
    {{0, 2}, {2, 1}} // target indices
);
```
instead of the recurring code blocks.
But since it is really just a minimal improvement and lots of typing to change. I probably wouldn't spend time on it. Just wanted to mention it.

Reply:
True, the reason I did it is to have something named, but those comments are as effective. Anyway, as you suggested, leaving it as it is